core-library/src/main/java/org/silverpeas/core/admin/persistence/InstanceDataTable.java

```diff
@@ -101,9 +108,20 @@ protected void prepareInsert(String insertQuery, PreparedStatement insert, Insta
     return params;
   }
 
-  private static final String SELECT_ALL_COMPONENT_PARAMETERS = "select "
-      + INSTANCEDATA_COLUMNS
-      + " from ST_Instance_Data where componentId = ? order by id";
+  /**
+   * Returns all component ids according to given param and param value
+   */
+  public List<Integer> getComponentIdsWithParameterValue(Parameter param) throws SQLException {
+    List<String> queryParams = new ArrayList<String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS2293)